sympy/core/power.py

```diff
@@ -555,12 +555,12 @@ def _eval_is_extended_real(self):
                 if self.exp.is_extended_nonnegative:
                     return True
             else:
-                if self.exp.is_integer:
+                if self.exp.is_integer and self.base.is_zero is not None:
```

Comment:
Here `self.base.is_zero` could be True, None or False. None means that it is possibly zero or possibly not. What is this check intended to do?  Should it be `and self.base.is_zero is False`?

Reply:
Yeah, I think it should be False, since by mentioning not None, there is a possibility that it could be True. But I am not able to figure out the fact that should it return True, even when self.base.is_zero is True??
@oscarbenjamin @smichr

Reply:
Consider it carefully by thinking about what `x**y` should be for different values of x and y. To understand the code you are working with you need to try and think through every possibility.

Reply:
In [1]: x=10
In [2]: print(x.is_finite)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-5852133bdaf6> in <module>
----> 1 print(x.is_finite)
**AttributeError: 'int' object has no attribute 'is_finite'**

Should this error come? 
**It is popping up in Sympy 1.5 developer version but not in Sympy 1.4.**

 @oscarbenjamin @smichr

Reply:
`x = 10` is a Python `int` that has no attribute `is_finite`. That has nothing to do with SymPy.
```
Python 2.7.12 (default, Nov 12 2018, 14:36:49) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> x = 10
>>> print(x.is_finite)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'int' object has no attribute 'is_finite' 
```
You can get a SymPy integer by invoking `Integer`.
```
>>> from sympy import Integer
>>> x = Integer(10)
>>> print(x.is_finite)
True
```

Reply:
Thanks, got it.

Reply:
Sorry. Instead of return True in line 559 return `not self.base.is_zero` because we know it's not None and if it is zero the answer is `zoo` which is not real; and if it is not zero then it *is* real.